### ISSUE TITLE
Sound crackles / pops with DirectSound audio driver

### ISSUE THREAD

**Issue by prietveld** [flyinghead/flycast#425]
**_Platform / OS / Hardware_**

Windows 7x64 and Windows 10
Intel Core i7-975
Nvidia GTX 1080 Ti
Creative Sound Blaster X-Fi Titanium Fatal1ty Professional

**_Description of the Issue_**

With both v1.0 and 1.1, when using DirectSound the audio crackles and pops. It is immediately noticeable on the BIOS / boot screen. The amount seems to vary per game, but it is most noticeable during FMV. The sdl2 driver does not have this issue (though default volume is very loud and boomy), but it does have the occasional audio drop-out or hitch during gameplay.

Interestingly, the version of Flycast I had installed before v1.1 (which I unfortunately no longer have access to an cannot confirm the version of, though it wasn't 1.x) did not have this issue. Were there versions available for download prior to 1.0 (or between 1.0 and 1.1) that have been taken ofline? It did not yet have mouse capture is all I can confirm about it.

**_Debugging Steps Tested_**

Basically tried all settings I could imagine being even remotely related, with zero effect (other than the sdl2 backend, see above).

**_Logs Gathered_**

Let me know if further information is required.

**Comment by flyinghead** [comment]
There was a change not long before 1.0 was released to the DirectSound driver: it now uses a dedicated thread and honors the Audio Latency setting. I suppose you were using a debug version that predates this change (builds are kept for 3 months).

That doesn't explain your issues though. I just tested on a low end win10 PC (i5-7260U @ 2.20GHz, RealTek audio) and there's no audio cracks and pops at all.

I assume you've tried changing Audio Latency in the audio settings. Do you have any audio device connected to your system, like DJ or audio equipment? Any audio app/plugin that could get in the way?
Some users have reported issues with some equipment and audio apps.

Are there settings on your sound card that could be related to the issue?

**Comment by flyinghead** [comment]
Unfortunately I don't have an offline archive of old builds.

You said the sdl2 default volume was "very loud and boomy." It should be at the same volume level as DirectSound as the audio signal is exactly the same.

**Comment by prietveld** [comment]
Not sure what to say. SDL2 is clearly louder, and to my ears a bit clipped because of it (but that could be subjective). Here are two recordings of Stupid Invaders, all I did in between was switch audio backends.

[recordings.zip](https://github.com/flyinghead/flycast/files/7747341/recordings.zip)

**Comment by flyinghead** [comment]
Very different sounds indeed. The SDL2 and DX drivers are probably using different routes or devices.

The DirectSound driver uses the system-wide default audio playback device. Not sure which one SDL chooses by default.

**Comment by prietveld** [comment]
Different in software only then, as there is only one physical sound card present in my setup.
